adapters/alkimi/alkimi.go

```diff
@@ -0,0 +1,195 @@
+	for _, imp := range bidRequest.Imp {
+
+		var bidderExt adapters.ExtImpBidder
+		var extImpAlkimi openrtb_ext.ExtImpAlkimi
+
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := json.Unmarshal(bidderExt.Bidder, &extImpAlkimi); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		var bidFloorPrice floors.Price
+		bidFloorPrice.FloorMinCur = imp.BidFloorCur
+		bidFloorPrice.FloorMin = imp.BidFloor
+
+		if len(bidFloorPrice.FloorMinCur) > 0 && bidFloorPrice.FloorMin > 0 {
+			imp.BidFloor = bidFloorPrice.FloorMin
+		} else {
+			imp.BidFloor = extImpAlkimi.BidFloor
+		}
+		imp.Instl = extImpAlkimi.Instl
+		imp.Exp = extImpAlkimi.Exp
+
+		temp := extObj{AlkimiBidderExt: extImpAlkimi}
+		temp.AlkimiBidderExt.AdUnitCode = imp.ID
+
+		extJson, err := json.Marshal(temp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		imp.Ext = extJson
+		updatedImps = append(updatedImps, imp)
+	}
+
+	return updatedImps, errs
+}
+
+func buildBidderRequest(adapter *adapter, encoded []byte) *adapters.RequestData {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	reqBidder := &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    encoded,
+		Headers: headers,
+	}
+	return reqBidder
+}
+
+// MakeBids will parse the bids from the Alkimi server
+func (adapter *adapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	if request == nil || response == nil || adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResp openrtb2.BidResponse
+	err := json.Unmarshal(response.Body, &bidResp)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	bidCount := len(bidResp.SeatBid)
+	if bidCount > 0 {
+		bidResponse := adapters.NewBidderResponseWithBidsCapacity(bidCount)
+		for _, seatBid := range bidResp.SeatBid {
+			for _, bid := range seatBid.Bid {
+				copyBid := bid
+				resolveMacros(&copyBid)
+				impId := copyBid.ImpID
+				imp := request.Imp
+				bidType, err := getMediaTypeForImp(impId, imp)
+				if err != nil {
+					errs = append(errs, err)
+				}
+				bidderBid := &adapters.TypedBid{
+					Bid:     &copyBid,
+					BidType: bidType,
+					Seat:    "alkimi",
+				}
+				bidResponse.Bids = append(bidResponse.Bids, bidderBid)
+			}
+		}
+		return bidResponse, errs
+	}
+	return nil, nil
+}
+
+func resolveMacros(bid *openrtb2.Bid) {
+	if bid == nil {
+		return
+	}
+	strPrice := strconv.FormatFloat(bid.Price, 'f', -1, 64)
+	bid.NURL = strings.Replace(bid.NURL, price_macro, strPrice, -1)
+	bid.AdM = strings.Replace(bid.AdM, price_macro, strPrice, -1)
```

Comment:
https://github.com/prebid/prebid-server/pull/3247#discussion_r1373389159

- should use  strconv.FormatFloat instead of fmt.Sprint
- may be error out if bid.Price is present and bid.NURL and bid.AdM has no PRICE_MACRO

Reply:
We use resolveMacros function like it in other adapters.  
Is it OK? OR should we modify this?

Reply:
> We use resolveMacros function like it in other adapters. Is it OK? OR should we modify this?

its okay your adapter can have your own implementation of `resolveMacros`